subprojects/launcher/src/main/java/org/gradle/launcher/cli/CommandLineActionFactory.java

```diff
@@ -135,6 +136,9 @@ public CommandLineParseFailureAction(CommandLineParser parser, Exception e) {
         }
 
         public void execute(ExecutionListener executionListener) {
+            if (!JavaVersion.current().isJava8Compatible()) {
+                System.out.println("Support for running Gradle using Java 7 has been deprecated and is scheduled to be removed in Gradle 5.0");
+            }
```

Comment:
:o: Unlike `BuildExceptionReporter` this probably can not produce a duplicated message. I would still remove this to not blow up the code with warning everywhere. I think it is fine not to issue the warning is Gradle is started in a broken way.

We could keep it, but then we need some test coverage for this as well.